keps/sig-network/4631-loadbalancerstatus/README.md

```diff
@@ -0,0 +1,1210 @@
+might look something like:
+
+```yaml
+status:
+  loadBalancer:
+    ingress:
+      - ip: 1.2.3.4
+    unimplementedFeatures:
+      - trafficDistribution
+```
+
+While at first glance, this is more user-friendly since it is explicit
+about what wasn't implemented, it runs into problems given its
+auto-generated-ness; since the provider doesn't know *anything* about
+these fields, it may report that it has failed to implement them even
+when that failure has no actual consequences (e.g., claiming to have
+not implemented `allocateLoadBalancerNodePorts` for a Service which is
+`allocateLoadBalancerNodePorts: true`). This in turn makes it *less*
+user-friendly (though this is not a problem for automated use from e2e
+tests, since the tests would know when it does and doesn't matter that
+a particular feature was unimplemented).
+
+#### Option 5: Make the Cloud Provider Respond to an Explicit User-Provided Set of Required Features
+
+Another alternative is that if the user actually wants to know, for
+sure, that the load balancer implements a particular feature, they can
+request that explicitly:
+
+```
+# The load balancer must be able to support mixed TCP and UDP, and
+# must support KEP-4444
+requiredFeatures:
+  - TCP
+  - UDP
+  - TrafficDistribution
+```
+
+Then the cloud provider could scan the list to see if there are any
+features it either doesn't implement or simply doesn't recognize, and
+refuse to create a load balancer if so.
+
+In that case, the e2e tests can just always require the specific
+features that they are testing, and skip themselves if the cloud
+provider returns a `LoadBalancerServing` with the value `False` and a
+`Reason` of `UnsupportedRequiredFeature` or the like.
+
+(This feature would also be useful in other cases: for example, if the
+user wants a load balancer using `LoadBalancerSourceRanges` and wants
+to ensure that the cloud provider doesn't create a load balancer if it
+can't correctly implement the source range filtering.)
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### `Service.Status.Conditions`
+
+The problem of indicating provisioning progress and provisioning
+failure can easily be solved with conditions on the Service.
+
+```
+<<[UNRESOLVED per-ingress conditions ]>>
+
+Would it be better to add a `Conditions` field to
+`v1.LoadBalancerIngress` so that we can specify conditions per element
+of `.Status.LoadBalancer.Ingress`? IOW, should it be possible for a
+load balancer to express that it has multiple IPs in different states?
+
+<<[/UNRESOLVED]>>
+```
+
+#### The `LoadBalancerServing` Condition
```

Comment:
Can we arrange it so that, as much as possible, the behavior and names are the same between Services and Gateways?